src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/TeleporterSystem.java

```diff
@@ -45,7 +33,11 @@
     @In
     EntityManager entityManager;
 
-    private Random random = new Random();
+    private final Random random = new Random();
+
+    private final int maxTeamSizeDifference = 3;
+    private int redTeamCount;
+    private int blackTeamCount;
```

Comment:
This is state that should be kept in a component - I think there is some game state entity that holds information relevant to a "game" of LaS.

Reply:
Actually game state information is scattered throughout the code as variables (probably should keep track of team scores through the ScoreComponent rather than variables) and components which are associated with separate entities . So for the time being I could create components for the respective teamCounts and attach them to their corresponding teleporterBlocks. Maybe later on move everything to a gameState entity which keeps track of all the game information if that is preferred.